src/app/lists/editModal.tsx

```diff
@@ -0,0 +1,142 @@
+    }
+
+    return (
+        <Modal header="Edit List" headerId="editList" isOpen={data !== undefined} onClose={onClose}>
+            <ModalInner>
+                <h3>Description</h3>
+                <TextInput
+                    defaultValue={toSubmit.item_name ?? ""}
+                    onChange={(event) => {
+                        const text = event.target.value as string;
+                        setToSubmit({ ...toSubmit, item_name: text });
+                    }}
+                    helperText="Description"
+                />
+                {headersAndTooltips.map(([[key, label], [tooltipKey]]) => {
+                    return (
+                        <DisplayContents key={key}>
+                            <h3>{label}</h3>
+                            <DataWithTooltipDiv>
+                                <TextInput
+                                    defaultValue={toSubmit[key] ?? ""}
+                                    helperText="Quantity"
+                                    onChange={(event) => {
+                                        const text = event.target.value as string;
+                                        setToSubmit({ ...toSubmit, [key]: text });
+                                    }}
+                                />
+                                <TextInput
+                                    defaultValue={toSubmit[tooltipKey] ?? ""}
+                                    helperText="Notes"
+                                    onChange={(event) => {
+                                        const text = event.target.value as string;
+                                        setToSubmit({ ...toSubmit, [tooltipKey]: text });
+                                    }}
+                                />
+                            </DataWithTooltipDiv>
+                        </DisplayContents>
+                    );
+                })}
+                <Button
+                    variant="contained"
+                    color="primary"
+                    onClick={async () => {
+                        let operation;
+
+                        if (data === null) {
+                            operation = supabase.from("lists").insert(toSubmit);
+                        } else {
+                            operation = supabase
+                                .from("lists")
+                                .update(toSubmit)
+                                .eq("primary_key", toSubmit.primary_key)
+                                .select();
+                        }
+
+                        const response = await operation;
+
+                        if (Math.floor(response.status / 100) !== 2) {
+                            setErrorMsg(
+                                `${response.status}: ${response.statusText} -- data: ${response.data}`
+                            );
+                        } else {
+                            window.location.reload();
```

Comment:
do you think it's worth commonising setting an error message from supabase in `supabase.ts`?
(again reload too)